ansible/playbooks/adhoc/hotfix/roles/update_node/tasks/main.yml

```diff
@@ -18,42 +15,41 @@
   changed_when: False
 
 # ansible will return a '17' if a match is found. '-1' means 'not found'
-- name: Set update_successful=True if upgrade succeeded
+- name: Set wanted_version_installed=True if upgrade succeeded
   set_fact: 
-    update_successful: True
+    wanted_version_installed: True
   when: "{{ post_upgrade.stdout.find(version_wanted) != -1 }}"
 
 - name: Debug node vars
-  debug: msg="update_successful={{update_successful}}, rpm_installed={{post_upgrade.stdout}}, version_wanted={{version_wanted}}"
+  debug: msg="wanted_version_installed={{wanted_version_installed}}, rpm_installed={{post_upgrade.stdout}}, version_wanted={{version_wanted}}"
 
 - name: Upgrade failed, so try a downgrade
   command: "yum -y downgrade atomic-openshift-{{ version_wanted }} atomic-openshift-clients-{{ version_wanted }} atomic-openshift-node-{{ version_wanted }} tuned-profiles-atomic-openshift-node-{{ version_wanted }} atomic-openshift-sdn-ovs-{{ version_wanted }}"
-  when: "{{ update_successful == False }}"
+  when: "{{ wanted_version_installed == False }}"
 
 - name: Check if downgrade is successful
   command: "rpm -qa atomic-openshift"
   register: post_downgrade
   changed_when: False
 
 # ansible will return a '17' if a match is found. '-1' means 'not found'
-- name: Set downgrade_successful=True if downgrade succeeded
+- name: Set wanted_version_installed=True if downgrade succeeded
   set_fact: 
-    downgrade_successful: True
-    upgrade_successful: True
+    wanted_version_installed: True
   when: "{{ post_downgrade.stdout.find(version_wanted) != -1 }}"
 
 - name: Alert if update is not successful, but continue
   debug:
     msg: "[WARNING] The desired rpm version does not match installed version! Upgrade and downgrade failed!"
-  when: "{{ update_successful == False and downgrade_successful == False }}"
+  when: "{{ wanted_version_installed == False }}"
   failed_when: True
   ignore_errors: True
 
 - name: Restart atomic-openshift-node service if upgrade was successful
   service:
     name: atomic-openshift-node
     state: restarted
-  when: "{{ update_needed == True and update_successful == True }}" 
+  when: "{{ update_needed == True and wanted_version_installed == True }}"
 
 - name: Ensure node service is running
   service:
```

Comment:
So how Ansible works is that each host specified in a playbook executes separately.

So failure of 1 node won't fail the other nodes. They will each execute independently.

So I'd definitely prefer we don't ignore_errors here.

As far as oadm, that makes more sense. I'd prefer it fail overall if it isn't able to set scheduling=true, but this is likely fine.